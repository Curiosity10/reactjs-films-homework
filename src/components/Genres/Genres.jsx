import React from 'react';
import PropTypes from 'prop-types';
import styles from './Genres.scss';

const Genres = (props) => {
  const { genres, fontColor, fontSize } = props;
  const fontStyle = {
    color: fontColor,
    fontSize,
  };
  return (
    genres.map(genre => (
      <span style={fontStyle} key={genre.id} className={styles.genre}>
        {genre.name}
      </span>
    )));
};

Genres.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.object).isRequired,
  color: PropTypes.string,
  fontSize: PropTypes.string,
};

export default Genres;
