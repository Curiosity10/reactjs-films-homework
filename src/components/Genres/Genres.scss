.genre {
  font-size: 0.8rem;
  margin-right: 3%;
  font-family: "Helvetica Neue Medium", Arial, Helvetica, sans-serif;
}
