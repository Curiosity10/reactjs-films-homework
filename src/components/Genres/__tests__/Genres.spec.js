import React from 'react';
import { create } from 'react-test-renderer';
import Genres from '..';
import { genres } from '../../../__mocks__/genres.json';

describe('Genres component renders correctly', () => {
  it('Genres renders correctly', () => {
    const tree = create(<Genres genres={genres} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
