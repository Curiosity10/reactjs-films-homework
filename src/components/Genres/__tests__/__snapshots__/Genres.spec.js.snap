// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genres component renders correctly Genres renders correctly 1`] = `
Array [
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Action
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Adventure
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Animation
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Comedy
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Crime
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Documentary
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Drama
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Family
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Fantasy
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    History
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Horror
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Music
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Mystery
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Romance
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Science Fiction
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    TV Movie
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Thriller
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    War
  </span>,
  <span
    className="genre"
    style={
      Object {
        "color": undefined,
        "fontSize": undefined,
      }
    }
  >
    Western
  </span>,
]
`;
